### PR TITLE
Gdrcopy prefix

### PR DESCRIPTION
**Pull request by cliffburdick**
## Pull Request Description
GDRCopy uses `prefix` as part of make and not `PREFIX`

## Author Checklist
* [x] Updated documentation (`pydocmd generate`) if any docstrings have been modified
* [x] Passes all unit tests

### CODE REVIEW

**Comment by samcmill** [comment]
It looks like the `prefix` vs `PREFIX` change straddles 2.1 -> 2.2.  The variable name will need to be conditional on the version.

**Comment by samcmill** [comment]
Also, the unit tests will need to be updated.

(I'm not sure why they're not getting triggered automatically.)

**Comment by cliffburdick** [comment]
> Also, the unit tests will need to be updated.
> 
> (I'm not sure why they're not getting triggered automatically.)

Thanks @samcmill, I'll update the pr and unit test

**Comment by samcmill** [comment]
BTW, here's an example for handing version dependent things: https://github.com/NVIDIA/hpc-container-maker/blob/master/hpccm/building_blocks/netcdf.py#L232

File: hpccm/building_blocks/gdrcopy.py [hunk 1]
```diff
@@ -101,7 +101,13 @@ def __init__(self, **kwargs):
             # CFLAGS is derived from COMMONCFLAGS, so rename.  See
             # https://github.com/NVIDIA/gdrcopy/blob/master/src/Makefile#L9
             make_opts['COMMONCFLAGS'] = make_opts.pop('CFLAGS')
-        make_opts['PREFIX'] = self.__prefix
+
+        # Version 2.2 changed the flag to lowercase prefix
+        if LooseVersion(self.__version) >= LooseVersion('2.2'):
```
Comment (samcmill) [inline hpccm/building_blocks/gdrcopy.py:106]: You probably need to import `LooseVersion` from `distutils.version`

File: test/test_gdrcopy.py [hunk 2]
```diff
@@ -32,6 +32,49 @@ def setUp(self):
         """Disable logging output messages"""
         logging.disable(logging.ERROR)
 
+    @ubuntu
+    @docker
+    def test_defaults_ubuntu(self):
```
Comment (samcmill) [inline test/test_gdrcopy.py:37]: Please give these tests unique names.

File: test/test_gdrcopy.py [hunk 3]
```diff
@@ -32,6 +32,49 @@ def setUp(self):
...
+        """Default gdrcopy building block"""
+        g = gdrcopy()
+        self.assertEqual(str(g),
+r'''# GDRCOPY version 2.2
```
Comment (samcmill) [inline test/test_gdrcopy.py:41]: The default version is still 2.1, right?  I don't think this test will pass as-is.

Comment (cliffburdick) [reply test/test_gdrcopy.py:41]: Thanks @samcmill. Should I keep around all the old tests for 1.3 and 2.1? I notices some things are tested in 2.1 and not 1.3, and vice versa.

Comment (samcmill) [reply test/test_gdrcopy.py:42]: The "default" tests are intended to call the building block with no options (i.e., just the defaults).  If you're going to bump the default version to 2.2 (probably a good idea), then these tests should reflect that.  In addition, add a new 2.1 specific test that sets `version='2.1'` to cover the 2.1 code branch.

**Comment by samcmill** [comment]
It looks like there was another change from 2.1 -> 2.2.  The library directory is now `lib` instead of `lib64`.  Can you please also make this conditional on the version?

File: test/test_gdrcopy.py [hunk 4]
```diff
@@ -23,6 +23,7 @@
 import unittest
 
 from helpers import centos, docker, ubuntu
+from distutils.version import LooseVersion
```
Comment (samcmill) [inline test/test_gdrcopy.py:26]: This should go in the building block, not the test.

Comment (cliffburdick) [reply test/test_gdrcopy.py:26]: Thanks @samcmill . Is there a way I can run the tests without pushing new changes to you each time?

Comment (cliffburdick) [reply test/test_gdrcopy.py]: Nevermind, I see it's running now!

File: hpccm/building_blocks/gdrcopy.py [hunk 5]
```diff
@@ -83,16 +84,8 @@ def __init__(self, **kwargs):
         self.__ospackages = kwargs.pop('ospackages', ['make', 'wget'])
         self.__prefix = kwargs.pop('prefix', '/usr/local/gdrcopy')
         self.__toolchain = kwargs.pop('toolchain', toolchain())
-        self.__version = kwargs.pop('version', '2.1')
+        self.__version = kwargs.pop('version', '2.2')
```
Comment (samcmill) [inline hpccm/building_blocks/gdrcopy.py:87]: Can you please update the docstring to reflect the version bump?

If you don't have `pydocmd`, I can regenerate the docs myself.

**Comment by samcmill** [comment]
Looks good.  Sorry this turned into a bigger can of worms than it initially looked.

**Pull request merged**